### DISCUSSION TITLE
Set Autolaunch in Preferable browser

### DISCUSSION THREAD

**Discussion by tmprabubiz** [lllyasviel/stable-diffusion-webui-forge#1911] [Q&A]
How to set the Autolaunch in a preferable browser.

I have Chrome as my default browser. But I want my Forge web UI interface to be with Firefox or if there is any other good browser suitable for Forge.

Thank You in advance

**Comment by zeittresor** [comment]
Hi. What about to just start firefox using http://127.0.0.1:7860/ as a startpage or as a bookmark in the top menu instead of using the autolaunch option ?

or do you mean how you can change the default browser using windows ? in that case just click start then type "default" -> enter -> select "webbrowser" change it to firefox.

..or just use something like this as a python replacement for the default autolaunch option (ex. for windows users):

import webbrowser
url = 'http://127.0.0.1:7860/'
firefox_path = 'C:/Program Files/Mozilla Firefox/firefox.exe'
webbrowser.get(firefox_path + ' %s').open(url)

**Reply by tmprabubiz** [reply]
Bookmarking the URL with FIrefox does create problems because I use Fooocus, ComfyUI and Ruined Fooocus in my windows pc. THe bookmarked URLs get to clash with each other and none open at times. Hence, I use Firefox as my default browser while using web UIs and I need to change the default to Chrome when I am trying the password reset links, and while connecting the Zoom calls. I could not work with things as a reflex, instead I need to be conscious as if I am working with two computers with different OS.
--------
And thank you for the response.
Do you mean to say that I should add the following lines in webui-user.bat
-----
import webbrowser
url = 'http://127.0.0.1:7860/'
firefox_path = 'C:/Program Files/Mozilla Firefox/firefox.exe'
webbrowser.get(firefox_path + ' %s').open(url)
-----
Please confirm. (It is 2.30am here, I may check your answer by tomorrow forenoon. Plz don't mind if I am responding you after that)

**Reply by JinJomes** [reply]
Set Firefox to open previous Tabs at Startup, open a tab to http://127.0.0.1:7860 as normal, close Firefox. Start ComFooo-Orge of your choice, open Firefox. Worst-Case, you'll have to refresh page to shake off whichever one you used last.

**Reply by tmprabubiz** [reply]
Thank You for the response, inspite of my repeated doubts
.
But I could not understand whatever did you said.
.
I want Google Chrome as my default browser for my routine - I had it already
.
I want my Ai web UI's like Forge, Fooocus, Comfy UI and Ruined Fooocus to use Firefox as the interface.
.
Do you have any solution to solve this issue exactly.
.
Thank You for your time

**Reply by a3nima** [reply]
disable autolaunch of forge and set the start page of firefox to http://127.0.0.1:7860/ (firefox default page to load when its started). Then you only have to start firefox after you have launched forge and you will be on the webUI page. its not fully automatical but it won't start in the "wrong" browser.

**Reply by tmprabubiz** [reply]
Wow, Thank You. It works
Simple and Great
I got it
